File: apps/desktop/src-tauri/src/api/providers.rs
use crate::api::ApiClient;
use crate::error::AppResult;
use crate::models::ProviderCatalogEntry;

pub async fn list_providers(client: &ApiClient) -> AppResult<Vec<ProviderCatalogEntry>> {
    client.get("/providers").await
}

pub async fn get_models(client: &ApiClient, provider_id: &str) -> AppResult<serde_json::Value> {
    client
        .get(&format!("/providers/{}/models", provider_id))
        .await
}

File: apps/console-macos/Sources/ConsoleCore/Api/Providers.swift
import Foundation

// Mirrors api/providers.rs

public extension ApiClient {
    func listProviders() async throws -> [ProviderCatalogEntry] {
        try await get("/providers")
    }

    func getModels(providerId: String) async throws -> JSONValue {
        try await get("/providers/\(providerId)/models")
    }
}
